clients/csharp-uwp/docs/sample-app-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+
+Code path in the sample app:
+- On first run MainPage.xaml.cs completes keyword registration, Background Service registration, mic verification, and voice activation setting verification.
+- On 1st stage detection, DialogManager.HandleSignalDetected is called, either through App.xaml.cs's OnBackgroundActivated or the ConversationalAgentSession.SignalDetected event handler in AgentSessionManager
+- HandleSignalDetected initializes the AgentAudioInputProvider with audio from the ConversationalAgentSession, creates a DirectLineSpeechDialogBackend instance, and passes the AgentAudioInputProvider through SetAudioSource
+- DialogManager listens for speech recognition events from the DirectLineSpeechDialogBackend or a timeout event from SignalDetectionHelper and surfaces the events to the UI
+- On keyword confirmation, the app uses the ConversationalAgentSession to request foreground activation and display UI
+
+## Bot interaction
+The IDialogBackend interface was designed to allow the use of any dialog service with the other components in the UWP Voice Assistant sample. It contains events for keyword signals, speech recognition output, and generic output in the form of the DialogResponse class. 
+
+The sample app's implementation of this interface, DirectLineSpeechDialogBackend, uses the Speech SDK and Direct Line Speech which provide an easy way to build a dialog service and a straightforward library to interface with it.
+
+The following is a walkthrough of the code path to process an utterance from the user while the app is listening followed by a text and audio response from the bot, altogether referred to as a "turn".
+
+### Turn during first activation
+While completing 3rd stage keyword verification, Direct Line Speech will begin to convert the activation audio into text. This reduces latency between keyword confirmation and the bot response. In practice, this means that, after providing audio to the DialogServiceConnector, the expected result is a [DialogServiceConnector.SpeechRecognized event for the keyword confirmation](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DirectLineSpeechDialogBackend.cs#L127) followed by a [DialogServiceConnector.SpeechRecognized event with the text version of what the user said when they activated the voice agent](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DirectLineSpeechDialogBackend.cs#L131). Direct Line Speech also immediately sends this text to the configured bot and, upon receiving a response, converts the text response from the bot into audio before finally sending the full response back to the application through the [DialogServiceConnector.ActivityReceived](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DirectLineSpeechDialogBackend.cs#L148) event.
```

Comment:
This topic is indeed potentially confusing. Not sure what the best way to address it is. It's not guaranteed that you'll get a `SpeechRecognizing` reason with just the keyword text, even, especially if you're speaking fairly quickly with no pause.

Maybe something like:

`Hello`

| Step | `DialogServiceConnector` event | `ResultReason` | Description |
|---|---|---|---|
| 1 | `SpeechRecognizing` | `RecognizingKeyword` | Keyword confirmed locally |
| 2 | `SessionStarted` | | Audio begins flowing to the Speech Service
| 3 | `SpeechRecognized` | `RecognizedKeyword` | Keyword verified in the cloud |
| 4 | `SpeechRecognizing` | `RecognizingSpeech` | In-progress speech-to-text as the user is speaking |
| ... | * | * | Speech continues |
| 5 | `SpeechRecognized` | `RecognizedSpeech` | Final speech-to-text result that is also sent to the bot |